src/hotspot/share/runtime/objectMonitor.inline.hpp

```diff
@@ -32,23 +32,34 @@
 #include "oops/markWord.hpp"
 #include "runtime/atomic.hpp"
 #include "runtime/globals.hpp"
+#include "runtime/javaThread.inline.hpp"
 #include "runtime/lockStack.inline.hpp"
 #include "runtime/synchronizer.hpp"
+#include "runtime/threadIdentifier.hpp"
 #include "utilities/checkedCast.hpp"
 #include "utilities/globalDefinitions.hpp"
 
+inline int64_t ObjectMonitor::owner_from(JavaThread* thread) {
+  int64_t tid = thread->lock_id();
+  assert(tid >= 3 && tid < ThreadIdentifier::current(), "must be reasonable");
```

Comment:
Should the "3" be a named constant with a comment?

Reply:
Yes, changed to use ThreadIdentifier::initial().